beginner_source/ddp_fault_tolerance.rst

```diff
@@ -0,0 +1,212 @@
+`Introduction <ddp_series_intro.html>`__ \|\| `What is DDP <ddp_theory.html>`__ \|\| `Single-node
+Multi-GPU training <ddp_multigpu.html>`__ \|\| **Fault
+Tolerance** \|\| `Multi-node
+training <intermediate/ddp_multinode.html>`__ \|\| `mingpt training <intermediate/ddp_minGPT.html>`__
+
+
+Fault-tolerant Distributed Training with ``torchrun``
+=====================================================
+
+Authors: `Suraj Subramanian <https://github.com/suraj813>`__
+
+In distributed training, a single process failure can
+disrupt the entire training job. Since the susceptibility for failure can be higher here, making your training
+script robust is particularly important here. You might also prefer your training job to be *elastic* i.e. the ability to increase or decrease the number
+of processes in realtime while running the training job.
+
+PyTorch has a utility ``torchrun`` that provides fault-tolerance as well
+as elastic training. When a failure occurs, torchrun logs the errors and
+attempts to automatically restart all the processes from the last saved
+“snapshot” of the training job. 
+
+The snapshot saves more than just the model state; it can include
+details about the number of epochs run, optimizer states or any other
+mutable attribute of the training job necessary for its continuity.
+
+.. grid:: 2
+
+   .. grid-item-card:: :octicon:`mortar-board;1em;` What you will learn
+      :shadow: none
+
+      -  Launching multi-GPU training jobs with ``torchrun``
+      -  Saving and loading snapshots of your training job
+      -  Structuring your training script for graceful restarts
+
+   .. grid-item-card:: :octicon:list-unordered;1em;` Prerequisites
+      :shadow: none
+
+      * High-level `overview <ddp_theory.html>`__ of DDP
+      * Familiarity with `DDP code <ddp_multigpu.html>`__ 
+      * A machine with multiple GPUs (this tutorial uses an AWS p3.8xlarge instance)
+      * PyTorch `installed <https://pytorch.org/get-started/locally/>`__ with CUDA
+      
+
+
+View the code used in this video: https://github.com/suraj813/distributed-pytorch/blob/main/multigpu_torchrun.py
+
+
+.. raw:: html
+
+   <div style="margin-top:10px; margin-bottom:10px;">
+     <iframe width="560" height="315" src="https://www.youtube.com/embed/9kIvQOiwYzg" frameborder="0" allow="accelerometer; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

Comment:
Just curious, if this is hosted on PyTorch youtube channel or in ones personal file?

Reply:
On the PyTorch youtube channel